### ISSUE TITLE
BulkheadTest intermittently failing

### ISSUE THREAD

**Issue by tomas-langer** [helidon-io/helidon#5746]
I get the following exception on local env:
```
[INFO] Running io.helidon.nima.faulttolerance.BulkheadTest
[ERROR] Tests run: 3, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 5.169 s <<< FAILURE! - in io.helidon.nima.faulttolerance.BulkheadTest
[ERROR] io.helidon.nima.faulttolerance.BulkheadTest.testBulkheadWithError  Time elapsed: 5.002 s  <<< FAILURE!
org.opentest4j.AssertionFailedError: Unexpected exception type thrown ==> expected: <java.util.concurrent.ExecutionException> but was: <java.util.concurrent.TimeoutException>
        at org.junit.jupiter.api.AssertThrows.assertThrows(AssertThrows.java:65)
        at org.junit.jupiter.api.AssertThrows.assertThrows(AssertThrows.java:37)
        at org.junit.jupiter.api.Assertions.assertThrows(Assertions.java:3007)
        at io.helidon.nima.faulttolerance.BulkheadTest.testBulkheadWithError(BulkheadTest.java:199)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
...
...
...
Caused by: java.util.concurrent.TimeoutException
        at java.base/java.util.concurrent.CompletableFuture.timedGet(CompletableFuture.java:1960)
        at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2095)
        at io.helidon.nima.faulttolerance.BulkheadTest.lambda$testBulkheadWithError$12(BulkheadTest.java:200)
        at org.junit.jupiter.api.AssertThrows.assertThrows(AssertThrows.java:55)
```

Problem is caused by a race condition when a task is finished (if another task enqueues in parallel, we may:
- enqueue another task
- release the semaphore permit
- never release the enqueued task
